src/commands/aksClusterProperties/aksClusterProperties.ts

```diff
@@ -19,37 +19,112 @@ export default async function aksClusterProperties(
       return;
     }
 
-    const extensionPath = getExtensionPath();
-    if (failed(extensionPath)) {
-      vscode.window.showErrorMessage(extensionPath.error);
+    await prepareClusterProperties(cluster.result);
+}
+
+async function prepareClusterProperties(
+  cloudTarget: AksClusterTreeItem
+): Promise<void> {
+  const clustername = cloudTarget.name;
+
+  const clusterData = await longRunning(`Loading ${clustername} cluster properties.`, async () => await getClusterData(cloudTarget));
+  if (failed(clusterData)) {
+    vscode.window.showErrorMessage(clusterData.error);
+    return;
+  }
+
+  const clusterState = await longRunning(`Determine ${clustername} cluster state.`, async () => await determineClusterState(cloudTarget, clustername));
+  if (!clusterState) {
       return;
-    }
+  }
 
-    await loadClusterProperties(cluster.result, extensionPath.result);
+  await loadWebViewClusterProperties(cloudTarget, clusterData.result, clusterState);
 }
 
-async function loadClusterProperties(
+async function loadWebViewClusterProperties(
     cloudTarget: AksClusterTreeItem,
-    extensionPath: string) {
+    clusterInfo: ClusterARMResponse,
+    clusterState: string
+) {
 
     const clustername = cloudTarget.name;
-    await longRunning(`Loading ${clustername} cluster properties.`,
-      async () => {
-        const clusterInfo = await getClusterProperties(cloudTarget, clustername);
-        if (failed(clusterInfo)) {
-          vscode.window.showErrorMessage(clusterInfo.error);
-          return;
-        }
 
+    const extensionPath = getExtensionPath();
+    if (failed(extensionPath)) {
+      vscode.window.showErrorMessage(extensionPath.error);
+      return;
+    }
+
+    await longRunning(`Loading webview ${clustername} for cluster properties.`,
+      async () => {
+        let clusterData: ClusterARMResponse | undefined;
         const webview = createWebView('AKS Cluster Properties', `AKS properties view for: ${clustername}`);
 
-        webview.html = getWebviewContent(clusterInfo.result, extensionPath);
+        webview.onDidReceiveMessage(
+          async (message) => {
+              if (message.command === "startCluster") {
+                clusterData = await onRecievePerformOperations(cloudTarget, clustername, 'start');
+              } else if (message.command === "stopCluster") {
+                clusterData = await onRecievePerformOperations(cloudTarget, clustername, 'stop');
+              }
+              const clusterState = await determineClusterState(cloudTarget, clustername);
+              if (!clusterState || !clusterData) {
+                  return;
```

Comment:
This should 'never' happen, right? If we end up here, we don't actually have the required information to display cluster properties correctly. So, in that case, should we close the webview and maybe show an error?